### PR TITLE
fix(structlog): appends to `default_processors` via list to avoid TypeError

### LINKED ISSUE #7665: Exception on certain structlog configuration
### Summary of problem
This bug is introduced by https://github.com/DataDog/dd-trace-py/pull/7341
Only happen when DD_LOGS_INJECTION is enabled and structlog is configured with a tuple of processors.

structlog [doc](https://www.structlog.org/en/stable/configuration.html) suggested that the `processors` keyword argument takes an [Iterable](https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable) of callables that act as processors. Tuple is a valid Iterable.
### Which version of dd-trace-py are you using?
2.2.0
### Which version of pip are you using?
N/A

### Which libraries and their versions are you using?
structlog==23.1.0
<details>
  <summary>`pip freeze`</summary>
  
</details>

### How can we reproduce your problem?
Minimal reproducible code:
```
import os
os.environ["DD_LOGS_INJECTION"] = "true"

import ddtrace
ddtrace.patch_all()

import structlog
structlog.configure(processors=(structlog.stdlib.add_log_level, structlog.stdlib.add_logger_name))
logger = structlog.getLogger(__name__)
logger.error("this line is never printed.")
```
### What is the result that you get?
```
Traceback (most recent call last):
  File ".../ddtrace_structlog.py", line 9, in <module>
    logger = structlog.getLogger(__name__)
  File ".../ddtrace/2/2/0/dist/lib/python3.9/ddtrace/contrib/structlog/patch.py", line 54, in _w_get_logger
    structlog._config._CONFIG.default_processors = dd_processor + structlog._config._CONFIG.default_processors
TypeError: can only concatenate list (not "tuple") to list

```
### What is the result that you expected?
The string `this line is never printed.` should be printed on console.

### PR DESCRIPTION
**Pull request by zarirhamza**
Fixes #7665 by changing `default_processors` to list accounting for all types of iterables before appending

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: releasenotes/notes/fix_structlog_processor_list-037c90b391dbfbc2.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    structlog: Fix appending to ``default_processors`` by converting to list.
```
Comment (mabdinur) [inline releasenotes/notes/fix_structlog_processor_list-037c90b391dbfbc2.yaml:4]: Can we call out the error users see in their application.

Something like:
```suggestion
    structlog: Fixes BlahError raised when ddtrace log processor is configured.
```

Comment (mabdinur) [inline releasenotes/notes/fix_structlog_processor_list-037c90b391dbfbc2.yaml:4]: Can we call out the error users see in their application.

Something like:
```suggestion
    structlog: Fixes BlahError raised when ddtrace log processor is configured.
```

**Review by mabdinur — APPROVED** [review]

File: releasenotes/notes/fix_structlog_processor_list-037c90b391dbfbc2.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    structlog: Fixes TypeError raised when ddtrace log processor is configured with a tuple
```
Comment (majorgreys) [inline releasenotes/notes/fix_structlog_processor_list-037c90b391dbfbc2.yaml:4]:
```suggestion
    structlog: Fixes ``TypeError`` raised when ddtrace log processor is configured with a tuple
```

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
@zarirhamza can we add a regression test?

**Pull request merged**